Mathlib/Order/Hom/Lattice.lean

```diff
@@ -1586,6 +1586,48 @@ theorem symm_dual_comp (g : BoundedLatticeHom βᵒᵈ γᵒᵈ) (f : BoundedLat
 
 end BoundedLatticeHom
 
+/-! ### Prod -/
+
+namespace LatticeHom
+variable [Lattice α] [Lattice β]
+
+/-- Natural projection homomorphism from `α × β` to `α`. -/
+def fst : LatticeHom (α × β) α where
+  toFun := Prod.fst
+  map_sup' _ _ := rfl
+  map_inf' _ _ := rfl
+
+/-- Natural projection homomorphism from `α × β` to `β`. -/
+def snd : LatticeHom (α × β) β where
+  toFun := Prod.snd
+  map_sup' _ _ := rfl
+  map_inf' _ _ := rfl
+
+@[simp, norm_cast] lemma coe_fst : ⇑(fst (α := α) (β := β)) = Prod.fst := rfl
+@[simp, norm_cast] lemma coe_snd : ⇑(snd (α := α) (β := β)) = Prod.snd := rfl
+lemma fst_apply (x : α × β) : fst x = x.fst := rfl
+lemma snd_apply (x : α × β) : snd x = x.snd := rfl
+
+end LatticeHom
+
+/-! ### Pi -/
+
+namespace LatticeHom
+variable {ι : Type*} {α : ι → Type*} [∀ i, Lattice (α i)]
+
+/-- Evaluation as a lattice homomorphism. -/
+def evalLatticeHom (i : ι) : LatticeHom (∀ i, α i) (α i) where
```

Comment:
Why repeat `LatticeHom` in NS and name?

Reply:
I think the naming is all over the place here

Reply:
I think we standardized on `FooHom.fst` and `Pi.evalFooHom`?

Reply:
Uh, really? That's kind of a terrible in-between

Reply:
Please just tell me what I should use and I will do it

Reply:
> Please just tell me what I should use and I will do it

Please rename the Prod version to `FooHom.fst`, as you had them before. The Pi version is correctly named as you have it right now.

Reply:
> Uh, really? That's kind of a terrible in-between

I think now that we're in Lean 4 we could reconsider the Pi naming in the face of dot notation (which works on the `.fst` definitions); but that's not something to mix into this PR.